src/test/java/com/norconex/committer/googlecloudsearch/GoogleCloudSearchCommitterTests.java

```diff
@@ -255,26 +268,30 @@ public void initShouldStartIndexingService() {
   public void addShouldFailWhenRawUploadModeIsSelectedAndBinaryContentIsNotInValidBase64() {
     setupConfig.initConfig(new java.util.Properties());
     subject.loadFromXml(mockConfig);
-    thrown.expect(CommitterException.class);
-    thrown.expectMessage(
-        "Binary content field is missing or invalid. Please configure BinaryContentTagger"
-            + " and make sure the '"
-            + GoogleCloudSearchCommitter.FIELD_BINARY_CONTENT
-            + "' field is left untouched (e.g. watch out for KeepOnlyTagger)");
     subject.commitBatch(Arrays.asList(addOperation(URL, MIME_PDF, "Inv@lid", !CONTENT_ERROR)));
+    verify(appender, times(8)).doAppend(captor.capture());
+    List<LoggingEvent> loggingEvents = captor.getAllValues();
+    assertEquals("Exception caught while committing: " + URL, loggingEvents.get(4).getRenderedMessage());
+    assertEquals(
+      "Binary content field is missing or invalid. Please configure BinaryContentTagger"
+        + " and make sure the '"
+        + GoogleCloudSearchCommitter.FIELD_BINARY_CONTENT
+        + "' field is left untouched (e.g. watch out for KeepOnlyTagger)", loggingEvents.get(5).getRenderedMessage());
   }
 
   @Test
   public void addShouldFailWhenRawUploadModeIsSelectedAndBinaryContentFieldIsMissing() {
     setupConfig.initConfig(new java.util.Properties());
     subject.loadFromXml(mockConfig);
-    thrown.expect(CommitterException.class);
-    thrown.expectMessage(
-        "Binary content field is missing or invalid. Please configure BinaryContentTagger"
-            + " and make sure the '"
-            + GoogleCloudSearchCommitter.FIELD_BINARY_CONTENT
-            + "' field is left untouched (e.g. watch out for KeepOnlyTagger)");
     subject.commitBatch(Arrays.asList(addOperation(URL, MIME_PDF, null, !CONTENT_ERROR)));
+    verify(appender, times(8)).doAppend(captor.capture());
```

Comment:
verifying 'times(8)' may better be replaced with something like atLeastOnce(). It is too rigid in the current way.

Reply:
agreed. changed to atLeastOnce.